zebra-chain/src/transaction.rs

```diff
@@ -92,7 +92,9 @@ pub enum Transaction {
 
 impl Transaction {
     /// Iterate over the transparent inputs of this transaction, if any.
-    pub fn inputs(&self) -> impl Iterator<Item = &TransparentInput> {
+    pub fn inputs(
+        &self,
+    ) -> impl Iterator<Item = &TransparentInput> + DoubleEndedIterator + ExactSizeIterator {
```

Comment:
I didn't notice we were using an impl trait here, I think its better to just expose the type directly here. The standard library does the same thing. https://twitter.com/lcnr7/status/1292105855123546113 explains some of the advantages of using concrete types for iterators.

Reply:
This is a good idea, I went further and just returned slices in 2130d3c